vowpalwabbit/exploration.cc

```diff
@@ -0,0 +1,121 @@
+#include "exploration.h"
+
+#include <stdexcept>
+#include <algorithm>
+#include <numeric>
+#include <cstring>
+
+using namespace std;
+
+namespace exploration
```

Comment:
Structurally, maybe this should be in a .h file?  That makes it easier to use in C++.  

Then, when we want to bind across languages, we can just have a dummy .cc file.

Reply:
can move to .cc, why do we need a dummy .cc file?

Reply:
done. TODO: dummy .cc unclear

Reply:
(Let's deal with binding later)

Reply:
Nuke?

Reply:
comes back to the other question if we favor header only, with potential include/namespace pollution or .h/.cc files.